### PR TITLE
bpo-40755: Add missing multiset operations to Counter()

### PR DESCRIPTION
**Pull request by rhettinger**
https://bugs.python.org/issue40755

### CODE REVIEW

File: Lib/collections/__init__.py [hunk 1]
```diff
@@ -876,6 +888,92 @@ def __iand__(self, other):
...
+            >>> Counter(a=1, b=0).isequal(Counter(a=1))
+            True
+            >>> Counter(a=1, b=0) == Counter(a=1)
+            False
+
+        Logically equivalent to:  +self == +other
+        '''
+        if not isinstance(other, Counter):
+            other = Counter(other)
+        for elem in set(self) | set(other):
```
Comment (remilapeyre) [inline Lib/collections/__init__.py:908]: Would this be a bit more readable:

```python
for elem in set(self) | set(other):
    left = max(self[elem], 0)
    right = max(other[elem], 0)
    if left != right:
        return False
return True
```

?

Comment (rhettinger) [reply Lib/collections/__init__.py:908]: Also slower ;-)

File: Lib/collections/__init__.py [hunk 2]
```diff
@@ -712,12 +712,24 @@ def __repr__(self):
...
-    # however, that would not be the case for p=Counter(a=1, b=0)
-    # and q=Counter(a=1) where the dictionaries are not equal.
+    # When the multiplicities are all zero or one, multiset operations
+    # are guaranteed to be equivalent to the corresponding operations
+    # for regular sets.
+    #     Given counter multisets such as:
+    #         cp = Counter(a=1, b=0, c=1)
+    #         cq = Counter(c=1, d=0, e=1)
+    #     The corresponding regular sets would be:
+    #         sp = {'a', 'c'}
```
Comment (csabella) [inline Lib/collections/__init__.py:722]: s/sp/cp

File: Lib/collections/__init__.py [hunk 3]
```diff
@@ -712,12 +712,24 @@ def __repr__(self):
...
+    #         cq = {'c', 'e'}
```
Comment (csabella) [inline Lib/collections/__init__.py]: s/cq/sq

Comment (rhettinger) [reply Lib/collections/__init__.py]: Thanks for noticing.

**Pull request merged**